src/service/QuestionGenerator.js

```diff
@@ -0,0 +1,34 @@
+import { randomNumberInRange } from './randomNumberInRange';
+import {QuestionService} from './QuestionService';
+
+const questionService = new QuestionService()
+
+export class QuestionGenerator {
+
+    constructor(mode) {
+        this.mode = mode;
+        this.minPokeId = 1;
+        this.maxPokeId = 152;
+        this.askedQuestions = [];
+        this.askedQuestionsCount = 0;
+    };
+
+    async getNextQuestion() {
+        if (this.askedQuestionsCount === 30) {
+            return undefined;
+        };
+        this.askedQuestionsCount++;
+
+        const currentQuestionsArray = []
+        for (let i = 1; i <= 4; i++) {
+            currentQuestionsArray.push(randomNumberInRange(this.minPokeId, this.maxPokeId, this.askedQuestions));
+        };
+        this.askedQuestions = this.askedQuestions.concat(currentQuestionsArray);
```

Comment:
In my opinion it would be better to use `push()` instead of `concat()` here. 
Check: https://stackoverflow.com/questions/44572026/difference-between-concat-and-push

but both works fine 🤷